packages/grafana-ui/src/components/Forms/Button.tsx

```diff
@@ -0,0 +1,138 @@
+
+  switch (size) {
+    case 'sm':
+      padding = `0 ${theme.spacing.sm}`;
+      fontSize = theme.typography.size.sm;
+      iconDistance = theme.spacing.xs;
+      height = theme.height.sm;
+      break;
+
+    case 'md':
+      padding = `0 ${theme.spacing.md}`;
+      fontSize = theme.typography.size.md;
+      iconDistance = theme.spacing.sm;
+      height = `${theme.spacing.formButtonHeight}px`;
+      break;
+
+    case 'lg':
+      padding = `0 ${theme.spacing.lg}`;
+      fontSize = theme.typography.size.lg;
+      iconDistance = theme.spacing.sm;
+      height = theme.height.lg;
+      break;
+
+    default:
+      padding = `0 ${theme.spacing.md}`;
+      iconDistance = theme.spacing.sm;
+      fontSize = theme.typography.size.base;
+      height = theme.height.md;
+  }
+
+  switch (variant) {
+    case 'primary':
+      borderColor = theme.colors.blueShade;
+      background = buttonVariantStyles(theme.colors.blueBase, theme.colors.blueShade, theme.colors.white);
+      break;
+
+    case 'secondary':
+      borderColor = selectThemeVariant({ light: theme.colors.gray70, dark: theme.colors.gray33 }, theme.type);
+      const from = selectThemeVariant({ light: theme.colors.gray7, dark: theme.colors.gray15 }, theme.type) as string;
+      const to = selectThemeVariant(
+        {
+          light: tinycolor(from)
+            .darken(5)
+            .toString(),
+          dark: tinycolor(from)
+            .lighten(4)
+            .toString(),
+        },
+        theme.type
+      ) as string;
+      background = buttonVariantStyles(from, to, selectThemeVariant(
+        { light: theme.colors.gray25, dark: theme.colors.gray4 },
+        theme.type
+      ) as string);
+
+      break;
+
+    case 'destructive':
+      borderColor = theme.colors.redShade;
+      background = buttonVariantStyles(theme.colors.redBase, theme.colors.redShade, theme.colors.white);
+      break;
+  }
+
+  return {
+    button: css`
+      label: button;
+      display: inline-flex;
+      align-items: center;
+      font-weight: ${theme.typography.weight.semibold};
+      font-size: ${fontSize};
+      font-family: ${theme.typography.fontFamily.sansSerif};
+      line-height: ${theme.typography.lineHeight.sm};
+      padding: ${padding};
+      text-align: ${withIcon ? 'left' : 'center'};
+      vertical-align: middle;
+      cursor: pointer;
+      border: 1px solid ${borderColor};
+      height: ${height};
+      border-radius: ${theme.border.radius.sm};
+      ${background};
+
+      &[disabled],
+      &:disabled {
+        cursor: not-allowed;
+        opacity: 0.65;
+        box-shadow: none;
+      }
+    `,
+    iconWrap: css`
+      label: button-icon-wrap;
+      display: flex;
+      align-items: center;
+    `,
+    icon: css`
+      label: button-icon;
+      margin-right: ${iconDistance};
+      filter: brightness(100);
+    `,
+  };
+});
+
+export const Button: FC<AbstractButtonProps> = ({
+  renderAs,
+  theme,
```

Comment:
New buttons should not accept theme anymore. Theme should be retrieved in a component with `useTheme`. You can use https://www.typescriptlang.org/docs/handbook/utility-types.html#omittk to get rid of theme from `AbstractButtonProps`